main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -574,13 +575,62 @@ object ConstraintGen {
         (resTpe, resEff)
 
       case Expr.StructNew(sym, fields, region, tvar, evar, loc) =>
-        throw new RuntimeException("joe tbd")
+        val (instantiatedFieldTpes, structTpe, regionVar) = instantiateStruct(sym, root.structs)
+        val visitedFields = fields.map { case (k, v) => visitExp(v) }
+        val (fieldTpes, fieldEffs) = visitedFields.unzip
+        // struct type
+        c.unifyType(tvar, structTpe, loc)
+        // field types
+        fields.zip(fieldTpes).foreach {
+          case ((fieldSym, expr), fieldTpe1) =>
+            instantiatedFieldTpes.get(fieldSym) match {
+              case None => () // Field does not exist. A resolution error will already have been generated
```

Comment:
either just do the get directly or throw InternalCompilerException.

Here I think you can just directly apply instantiatedFieldTpes

Reply:
The reason I do this is that it is "valid" to get to this point with the field not existing, since the resolver generates a recoverable error when the field does not exist. Thus, if we have `new S {a = 3} @ r` if `a` does not exist, we would get to the constraint generation and don't want to crash the compiler

Reply:
Oh, I see.

Reply:
I think, where we detect the error, you should drop the extra field.

Reply:
Is that easily doable or more cumbersome?

Reply:
I think that's probably not too hard, but what if we have
`new S {a = "hello" + 3} @ r`

Wouldn't we want `a` to persist to the typer?

Reply:
OK, I agree with your points.

Now turn those ideas into a comment or two :)

Reply:
I added a comment - I can separate them up into 2 comments(one at unify and one at the `visitExp` if that would be better